backend/app/src/main/java/com/moh/phlat/backend/service/DbUtilityServiceImpl.java

```diff
@@ -209,13 +209,26 @@ public void loadProcessDataToPlr(Long controlTableId, String authenticatedUserId
 			Iterable<ProcessData> processDataList = processDataRepository
 					.getAllProcessDataByControlTableId(controlTableId);
 
+			// Begin load process
 			for (ProcessData processData : processDataList) {
-				// skip record marked as DO_NOT_LOAD and send to PLR VALID records only
+				// Skip record marked as DO_NOT_LOAD and send to PLR VALID records only
 				if (!"Y".equals(processData.getDoNotLoadFlag()) && RowStatusService.VALID.equals(processData.getRowstatusCode())) {
 					logger.info("loading process data with id: {} to PLR.", processData.getId());
 
+					// Load the ProcessData record
 					MaintainResults maintainResults = plrDataLoad.loadPlrViaEsb(control, processData);
-					for (PlrError plrError : maintainResults.getFacilityResult().getPlrErrors()) {
+					
+					// Set the RowStatusCode of the ProcessData record
+					if (maintainResults.isAllLoaded()) {
+						processData.setRowstatusCode(RowStatusService.COMPLETED);
+					} else if (maintainResults.hasAnyDuplicates()) {
+						processData.setRowstatusCode(RowStatusService.POTENTIAL_DUPLICATE);
+					} else {
+						processData.setRowstatusCode(RowStatusService.LOAD_ERROR);
+					}
+					
+					// Collect and add any error messages into the Messages table for display on the ProcessData record
+					for (PlrError plrError : maintainResults.getPlrErrors()) {
```

Comment:
This loop is redundant. Since process data entity has a 1-many relationship with messages, setting the messages in the process data and saving process data should save process data and messages also automatically.

Reply:
Gotcha, I didn't realize the Messages table could be populated through ProcessData like this. This actually cleans up the code in a couple ways. So I'm moving the Message object creation into the addError method of each response class and then removing this loop; on top of that, I am removing the PlrError class, since I am now directly making and inserting Message records into the ProcessData records and don't need this class to carry those details.